python_modules/libraries/dagster-dbt/dagster_dbt/cloud/resources.py

```diff
@@ -451,7 +451,12 @@ def run_job_and_poll(
         final_run_details = self.poll_run(
             run_id, poll_interval=poll_interval, poll_timeout=poll_timeout, href=href
         )
-        output = DbtCloudOutput(run_details=final_run_details, result=self.get_run_results(run_id))
+        try:
+            run_results = self.get_run_results(run_id)
+        # if you fail to get run_results for this job, just leave it empty
+        except Failure as _:
```

Comment:
Could we use the more specific failure here, and maybe `self._log.info` that no run results exist?
```suggestion
        except DagsterDbtCliOutputsNotFoundError:
```

Reply:
definitely agree on adding logging, but dagster.Failure is actually the most specific error we can get here (without a more involved change). This Failure exception comes from the underlying self.make_request() call, which will emit a Failure event if a request exception if encountered (after retrying a few times). We could pass in a flag through those calls telling make_request to propagate the original exception (rather than dagster.Failure), then just catch the specific RequestException here, but I figured that would be a pretty minor improvement vs. the added complexity.